ideascube/serveradmin/catalog.py

```diff
@@ -126,6 +129,10 @@ def remove(self, package):
         self._removed.append(package)
 
     def commit(self):
+        visible_packages = get_config('ideascube', 'displayed-package-ids')
+        [visible_packages.remove(p.id) for p in self._removed if p.id in visible_packages]
+        visible_packages.extend(p.id for p in self._installed)
+        set_config('ideascube', 'displayed-package-ids', visible_packages, User.objects.get_system_user())
```

Comment:
So with this commit, the base `Handler` essentially becomes coupled with the Ideascube DB.

I'm not sure I'm comfortable with that.

Wouldn't it be better to move this out of the `Handler.commit` method and into the `Catalog` instead?

The `Catalog` is a higher level thing. If we decided to split all this code out of Ideascube then the base `Handler` certainly wouldn't be able to access the Ideascube DB any more.

But moving this code to the `Catalog`, then Ideascube could extend the `Catalog` class to make it do those DB operations, and the base `Catalog` wouldn't have to know about the DB.

Does that make sense?